src/lib/mongoBackend/MongoCommonUpdate.cpp

```diff
@@ -297,25 +427,34 @@ static bool mergeAttrInfo(BSONObj& attr, ContextAttribute* caP, BSONObj* mergedA
   /* 1. Add value, if present in the request (it could be omitted in the case of updating only metadata).
    *    When the value of the attribute is empty (no update needed/wanted), then the value of the attribute is
    *    'copied' from DB to the variable 'ab' and sent back to mongo, to not destroy the value  */
-  if (caP->value != "" || caP->compoundValueP != NULL)
+  if (!emptyAttributeValue(caP))
   {
     valueBson(caP, ab);
   }
   else
   {
-    /* Slightly different treatment, depending on whether the attribute value in DB is compound or not */
-    if (attr.getField(ENT_ATTRS_VALUE).type() == Object)
+    /* Slightly different treatment, depending on attribute value type in DB (string, number, boolean, vector or object) */
+    switch (attr.getField(ENT_ATTRS_VALUE).type())
     {
-      ab.append(ENT_ATTRS_VALUE, attr.getField(ENT_ATTRS_VALUE).embeddedObject());
-    }
-    else if (attr.getField(ENT_ATTRS_VALUE).type() == Array)
-    {
-      ab.appendArray(ENT_ATTRS_VALUE, attr.getField(ENT_ATTRS_VALUE).embeddedObject());
-    }
-    else
-    {
-      ab.append(ENT_ATTRS_VALUE, STR_FIELD(attr, ENT_ATTRS_VALUE));
+      case Object:
+        ab.append(ENT_ATTRS_VALUE, attr.getField(ENT_ATTRS_VALUE).embeddedObject());
+        break;
+      case Array:
+        ab.appendArray(ENT_ATTRS_VALUE, attr.getField(ENT_ATTRS_VALUE).embeddedObject());
+        break;
+      case NumberDouble:
+        ab.append(ENT_ATTRS_VALUE, attr.getField(ENT_ATTRS_VALUE).Number());
+        break;
+      case Bool:
+        ab.append(ENT_ATTRS_VALUE, attr.getBoolField(ENT_ATTRS_VALUE));
+        break;
+      case String:
+        ab.append(ENT_ATTRS_VALUE, STR_FIELD(attr, ENT_ATTRS_VALUE));
+        break;
+      default:
```

Comment:
Looking to current version of the code, it seems to be fixed (although I'm not sure in which commit).